openapi/task_execution_service.openapi.yaml

```diff
@@ -654,6 +671,14 @@ components:
           type: string
           description: |-
             Optional user-provided description of task for documentation purposes.
+        callback_url:
+          type: string
+          format: uri
+          description: |-
+            Optional user-provided callback URL. Note that the URL can be
+            provided while creating a new task, but it MUST NOT be displayed when
```

Comment:
I think we should return the callback URL together with the rest of task payload. Other information contained in the task details can be equally sensitive. The only element to think about is whether it should be returned only in FULL or already in BASIC (probably BASIC makes more sense).